Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,398 @@
+            registry->Remove(bookmarkKey + "/");
+            AZ_Warning(
+                "SEditorSettings", saved, R"(Unable to save Editor Preferences registry file to path "%s"\n)",
+                editorBookmarkFilePath.c_str());
+        }
+
+        bool ViewBookmarkLoader::SaveBookmark(ViewBookmark bookmark, StorageMode mode)
+        {
+            switch (mode)
+            {
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Shared:
+                return SaveSharedBookmark(bookmark);
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Local:
+                return SaveLocalBookmark(bookmark);
+            case AzToolsFramework::ViewBookmarkLoaderInterface::Invalid:
+                return false;
+            }
+            return false;
+        }
+
+        bool ViewBookmarkLoader::SaveLastKnownLocationInLevel(ViewBookmark bookmark)
+        {
+            return SaveLocalBookmark(bookmark, true);
+        }
+
+        bool ViewBookmarkLoader::LoadViewBookmarks()
+        {
+            struct ViewBookmarkVisitor : AZ::SettingsRegistryInterface::Visitor
+            {
+                ViewBookmarkVisitor()
+                    : m_viewBookmarksKey{ "/O3DE/ViewBookmarks" } {};
+
+                AZ::SettingsRegistryInterface::VisitResponse Traverse(
+                    AZStd::string_view path,
+                    AZStd::string_view,
+                    AZ::SettingsRegistryInterface::VisitAction action,
+                    AZ::SettingsRegistryInterface::Type) override
+                {
+                    if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
+                    {
+                        // Strip off the last JSON pointer key from the path and if it matches the viewmark key then add an entry
+                        // to the ViewBookmark Map
+                        AZStd::optional<AZStd::string_view> localBookmarksID = AZ::StringFunc::TokenizeLast(path, "/");
+                        if (path == m_viewBookmarksKey && localBookmarksID && !localBookmarksID->empty())
+                        {
+                            AZStd::string_view bookmarkKey = localBookmarksID.value();
+
+                            auto existingBookmarkEntry = m_bookmarkMap.find(bookmarkKey);
+                            if (existingBookmarkEntry == m_bookmarkMap.end())
+                            {
+                                m_bookmarkMap.insert(AZStd::make_pair(bookmarkKey, AZStd::vector<ViewBookmark>{}));
+                            }
+                        }
+                    }
+
+                    return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+                }
+
+                using AZ::SettingsRegistryInterface::Visitor::Visit;
+                void Visit(
+                    AZStd::string_view path, AZStd::string_view valueIndex, AZ::SettingsRegistryInterface::Type, double value) override
```

Comment:
comment: This looks great but I think it would be awesome to consider how to test this, are there any existing examples of this @lumberyard-employee-dm? Do you have any recommendations? Thanks!

Reply:
Yes, I've seen a couple of similar Visitors implemented. Going to take a look at those

Reply:
Actually I added a "new hotness" for helping with visiting fields in the SettingsRegistry a few months back.

It is the [SettingsRegistryVisitorUtils](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryVisitorUtils.h).

It makes it easier to visit fields of array and objects without the need for the peeling of the last path of a json pointer path.

There are examples of how to use it in the UnitTest in [SettingsRegistryVisitorUtilsTests.cpp](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Settings/SettingsRegistryVisitorUtilsTests.cpp)

Reply:
If you need some examples of how to use it, I currently have some inprogress changes in a branch to leverage the SettingsRegistryVisitorUtils.

Here is a link to a file in my Gem Filesystem Alias Support branch: https://github.com/aws-lumberyard-dev/o3de/blob/396a821ec0db4e0dd8a6d110dc972edafbbd6285/Code/Framework/AzFramework/AzFramework/Gem/GemInfo.cpp#L27-L106